### PR TITLE
openshift-lightspeed-operator onboarding

### PR DESCRIPTION
**Pull request by vbelouso**
Add CI configuration to onboard the `openshift-lightspeed-operator` on Prow.

### CODE REVIEW

**Comment by vbelouso** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-operator-main-images

File: ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    name: release
+    namespace: openshift
+    tag: golang-1.21
+images:
+- dockerfile_path: Dockerfile
+  to: openshift-lightspeed-operator
+promotion:
+  to:
+  - name: "4.15"
+    namespace: lightspeed-operator
```
Comment (xrajesh) [inline ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml:13]: Verify if this is correct - light-speed service is using ols as namespace here .

Comment (raptorsun) [reply ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml:13]: I agree, let's publish the artifacts to the same as lightspeed-service repository does.

File: ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+releases:
+  initial:
+    integration:
+      name: "4.15"
+      namespace: ocp
+  latest:
+    integration:
+      include_built_images: true
```
Comment (raptorsun) [inline ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml:21]: Just a question: According to the [reference](https://steps.ci.openshift.org/ci-operator-reference#line1264),  it "include images built during the test itself", are we including the contents of the test images defined in 
"tests=>container=>from" (line 35) ?🤔

File: ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      name: "4.15"
+      namespace: ocp
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
```
Comment (raptorsun) [inline ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml:31]: An additional test to ensure all generated files are up to date can be useful:
```yaml
tests:
- as: generate
  commands: make generate --always-make && git diff --exit-code
  container:
    from: src
```

**Comment by raptorsun** [comment]
Just finished review while reading the long CI documents. Generally I do not see big issues, just 2 bits 😊
1. I suggest we change the promotion namespace to `ols` in accord with lightspeed-service repository
2. Adding a generate test 
The question on `include_built_images` is just out of curiosity, no modificaiton needed.

After we get the image building right, we can mirror the images to quay.io, as in [this PR for lightspeed-service repo](https://github.com/openshift/release/pull/45790).

File: ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml (hunk 1 above)
Comment (bparees) [reply ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml:13]: yes, please align this with lightspeed-service.

https://github.com/openshift/release/blob/1a80de2a855bae391be7e49018be32086076b9d8/ci-operator/config/openshift/lightspeed-service/openshift-lightspeed-service-main.yaml#L9-L12

File: ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml (hunk 2 above)
Comment (bparees) [reply ci-operator/config/openshift/lightspeed-operator/openshift-lightspeed-operator-main.yaml:21]: yeah i don't think we need or want this, @vbelouso can you explain the intent?

**Pull request merged**